lib/read4.g

```diff
@@ -23,3 +23,10 @@ ReadLib( "string.gi"   ); # since StringFile is needed early
 # for dealing with test files and manual examples
 ReadLib("test.gd");
 ReadLib("test.gi");
+
+if not IsBound(HPCGAP) then
+  # Galois group identification relies on the list of transitive groups
+  # (in HPC-GAP these file are read in read.g)
```

Comment:
But why is this different between GAP and HPC-GAP? The comment should explain that (which is the difficult, non-trivial part -- in contrast figuring out where GAP vs. HPC-GAP loads something can be figured with a simple "grep" in a few seconds)

And independently of that: Why does regular GAP put this into `read4.g`? I'd expect this to be in `read3.g` and `read5.g`, for the .gd and .gi respectively.

Reply:
@fingolfin: Because we want to please both installations with and without new packages. Starting GAP with `-D` option, one can see that the relative order in which these files are read is different. If there is no transgrp package, it is 
```
#I  READ_GAP_ROOT: loading 'trans/trans.gd' as GAP file
#I  READ_GAP_ROOT: loading 'trans/trans.grp' as GAP file
#I  READ_GAP_ROOT: loading 'trans/trans.gi' as GAP file
#I  READ_GAP_ROOT: loading 'lib/galois.gd' as GAP file
#I  READ_GAP_ROOT: loading 'lib/galois.gi' as GAP file
```
With it we have
```
#I  READ_GAP_ROOT: loading 'lib/trans.gd' as GAP file
#I  READ_GAP_ROOT: loading 'lib/galois.gd' as GAP file
#I    Read( "/Users/alexk/GITREPS/gap/pkg/transgrp/lib/trans.grp" )
#I    Read( "/Users/alexk/GITREPS/gap/pkg/transgrp/lib/trans.gi" )
#I  READ_GAP_ROOT: loading 'lib/galois.gi' as GAP file
```
Furthermore, `TRANSREGION` is defined in hpcgap/trans/trans.gd - when the package is present, this will not be read, so I define it in `hpcgap/lib/read.g`. 

I was testing this with packages and without package, and found this "equilibrium". It works now while we need to cater for both setups. As soon as we will go to the next stage and will remove `trans` and `hpcgap/trans`, I will look again and will try to converge handling of this in GAP and HPC-GAP.

Reply:
@alex-konovalov I still don't understand. You described the effect, telling me that the load order is different between GAP and HPC-GAP -- and, well, of course it is, as e.g. the `ReadLib` calls in a different order. But that still doesn't give me a logical reason for why we do it like that. 

My POV is this: Either there are strong reasons why it *must* be different, then we should document them in comments quite clearly. Or else, there are purely technical obstacles which we can and should remove, to make it possible to align this code.

The `TRANSREGION` sounds like the latter to me. Indeed, this is completely missing from the `transgrp` package -- so I guess before we can fix it, we must make `transgrp` HPC-GAP ready.

I'll look into that now.

Reply:
@fingolfin it's not only that the load order is different between GAP and HPC-GAP - it is also different dependently on whether the data library is a package or directory.

I have been trying to resolve all problems without making any changes in transgrp, assuming that those can be made eventually - transgrp is not hosted under gap-packages and has a longer release cycle. If you could look at that, please do, of course.

Reply:
But `transgrp` does not contain any HPC-GAP specific changes. So with this PR, support for transitive groups in HPC-GAP will be completely broken, no? And probably also for `smallgrp`, unless that contains HPC-GAP specific changes (I didn't see any on a quick glance just now... @markuspf ?)

So, for now it might be best to leave out the `LoadPackage` bits for HPC-GAP, until those packages contain the HPC-GAP specific changes, no?